### ISSUE TITLE
Crash when reading a rdc recorded from a crashed application

### ISSUE THREAD

**Issue by Extra-Creativity** [baldurk/renderdoc#3021]
## Description
**TL;DR: RenderDoc crash when reading an rdc recorded from a crashed application.**
Recently I'm trying to write an OpenGL application to produce soft shadows. But it crashes with weird behaviors, so I try to debug it by  RenderDoc to see what happens. I use "Queue Captures" and get the first 2 frames, but opening the second frame rdc will make RenderDoc crash suddenly too(the file size abnormally big too, about 20MB). I also try to save the file and reopen it in RenderDoc, but it still crashes. I'll attach the rdc file at the end.

If it's needed, the program can be seen [here](https://github.com/Extra-Creativity/OpenGLFramework/tree/v1.3/examples/complex-examples/SoftShadow-Advanced); These weird things happen on this program crash:
+ The OS freezes for a period of time;
    + On Ubuntu 20.04, though the cursor is movable, no action will receive responses(including trying to click other applications, or close the GLFW window of the OpenGL application), and even the console cannot be created by shortcut. After a long period of time, the desktop flashes and it's back to normal.
    + On Windows 10, the time is shorter(only freeze about 5 seconds), and the desktop will not flash.
+ After checking the debugger, it reports "Unhandled exception at 0x00007FFA43ECB935 (nvoglv64.dll) in SoftShadow-Advanced.exe: Fatal program exit requested".
+ After checking the profiler, it can be found that the program is stuck at glUniformMatrix4fv because it takes half of the program time abnormally. GPU usage will reach almost 100% at a time and slowly decrease until the program crashes. CPU usage is normal. The profiler also shows that OS kernel events take most of the time.
+ After checking the fragment shader, it can be found that accessing a local variable or not will make the program crash or not. But in fact there are many calling paths that use this function, and other paths are OK, so this may not be the root cause.
+ By outputting in the loop, it can be found that only 3 frames survive. The first frame will be stuck for a long time, and then quickly the following two frames, and then the whole program totally stops. But those frames are not rendered normally, too(i.e. only the background color can be seen in the GLFW window).

## Steps to reproduce
Run the program described above in RenderDoc, enable queue capture, and try to open the second rdc file.

## Environment

* RenderDoc version: v1.28
* Operating System: Windows 10
* Graphics API: OpenGL
* GPU: NVIDIA GTX 1650, driver version 31.0.15.1601

[temp.zip](https://github.com/baldurk/renderdoc/files/12326264/temp.zip)

**Comment by baldurk** [comment]
I'm not sure I understand what you're saying. Your program crashes even without RenderDoc present? if that's the case then RenderDoc will not likely be able to help with that and it strongly implies that your program is hitting invalid behaviour which RenderDoc explicitly does not support.

Does this problem reproduce with a valid program that shows no signs of problems when run independently? That would be the first step before investigating this as a RenderDoc bug.

**Comment by Extra-Creativity** [comment]
Ah, okay, my program crashes without RenderDoc and I try to debug with it.
